lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -549,12 +552,38 @@ func addHopHint(hopHints *[]func(*zpay32.Invoice),
 	)
 }
 
+// SelectHopHintsCfg contains the dependencies required to obtain hop hints
+// for an invoice.
+type SelectHopHintsCfg struct {
+	// IsChannelActive is used to generate valid hop hints.
+	IsChannelActive func(chanID lnwire.ChannelID) bool
+
+	// IsPublicNode is returns a bool indicating whether the node with the
+	// given public key is seen as a public node in the graph from the
+	// graph's source node's point of view.
+	IsPublicNode func(pubKey [33]byte) (bool, error)
+
+	// FetchChannelEdgesByID attempts to lookup the two directed edges for
+	// the channel identified by the channel ID.
+	FetchChannelEdgesByID func(chanID uint64) (*channeldb.ChannelEdgeInfo,
```

Comment:
I'm a bit hesitant here but do we need to use the `channeldb` types? Could we just create custom types similar to `HopHintInfo`?

Reply:
My reasoning for adding `HopHintInfo` is that `OpenChannel` has many exported fields like `Packager`/`RevocationKeyLocator` which need lnd internals, so it seemed cleaner to add a new struct with the bare minimum of what we need. 

`ChannelEdgePolicy` and `ChannelEdgeInfo` have far fewer of those, and we use most of their fields so I left them s is . Happy to add new structs if we think its a big improvement.